src/Processors/Formats/Impl/ParallelParsingInputFormat.cpp

```diff
@@ -147,6 +147,13 @@ void ParallelParsingInputFormat::onBackgroundException(size_t offset)
 
 Chunk ParallelParsingInputFormat::generate()
 {
+    /// Delayed launching of segmenting thread
+    if (!parsing_started.exchange(true))
```

Comment:
I suggest to wrap it into `unlikely()` to yield better branch prediction in common case.